apps/charge-api-service/src/bundler-api/bundler-api.interceptor.ts

```diff
@@ -94,14 +101,72 @@ export class BundlerApiInterceptor implements NestInterceptor {
     return requestConfig
   }
 
-  private prepareUrl (environment) {
+  private prepareUrl (environment, bundlerProvider) {
     if (isEmpty(environment)) throw new InternalServerErrorException('Bundler environment is missing')
-    const config = this.configService.get(`bundler.${environment}`)
+    const config = this.configService.get(`bundler.${bundlerProvider}.${environment}`)
 
     if (config.url) {
       return config.url
     } else {
       throw new InternalServerErrorException(`${capitalize(environment)} bundler environment is missing`)
     }
   }
+
+  private contructUserOp (context: ExecutionContext, requestConfig: AxiosRequestConfig, response) {
+    const request = context.switchToHttp().getRequest()
+    const bundlerProvider = request.query?.provider ?? BundlerProvider.ETHERSPOT
+    const param = requestConfig.data.params[0]
+    const base = { userOpHash: response?.result, apiKey: request.query.apiKey }
+
+    if (bundlerProvider === BundlerProvider.PIMLICO) {
+      return {
+        ...param,
+        ...base,
+        initCode: param.initCode ?? '0x',
+        sponsorId: param.paymaster ? BundlerProvider.PIMLICO : undefined,
+        paymasterAndData: param.paymasterData,
+        paymasterData: undefined
+      }
+    }
+    return { ...param, ...base }
+  }
+
+  private async isOperatorSponsoredQuotaExceeded (context: ExecutionContext, requestConfig: AxiosRequestConfig) {
```

Comment:
Please extract the isOperatorSponsoredQuotaExceeded logic into a dedicated service (or use an existing domain service) that handles operator validation and sponsorship quota checks. This keeps the interception layer lean and makes the logic reusable and easier to test.
This separation helps follow the single-responsibility principle and makes the quota logic easier to unit test in isolation.

Reply:
@NikolayRn extracted